qmk_storage.py

```diff
@@ -41,14 +42,16 @@
 )
 
 # Check to see if S3 is working, and if not print an error in the log.
-try:
-    s3.create_bucket(Bucket=S3_BUCKET)
-except Exception as e:
-    if e.__class__.__name__ not in ['BucketAlreadyOwnedByYou', 'BucketAlreadyExists']:
-        logging.warning('Could not contact S3! Storage related functionality will not work!')
+for bucket in [S3_BUCKET, COMPILE_S3_BUCKET]:
+    try:
+        s3.create_bucket(Bucket=bucket)
+
+    except Exception as e:
+        if e.__class__.__name__ not in ['BucketAlreadyOwnedByYou', 'BucketAlreadyExists']:
+            logging.warning('Could not contact S3! Storage related functionality will not work!')
```

Comment:
```suggestion
    
    except (botocore.exceptions.BucketAlreadyOwnedByYou, botocore.exceptions.BucketAlreadyExists) as e:
        logging.debug('Bucket already exists.')
    except Exception as e:
        logging.warning('Could not contact S3! Storage related functionality will not work!')
```

Reply:
I think [line 8](https://github.com/qmk/qmk_compiler/blob/public_firmware/qmk_storage.py#L8) could also be just `import botocore.exceptions`.

Reply:
Did you actually test this? When I first wrote the code I couldn't find anything that would actually match the exception as thrown.

Reply:
You are right, these exceptions are not directly exposed on the client side.
After digging through the docs, I found [the recommended way to catch these](https://botocore.amazonaws.com/v1/documentation/api/latest/client_upgrades.html#error-handling), so updated the comment.